File: src/main/java/commands/slashadapters/adapters/WarnRemoveAdapter.java
package commands.slashadapters.adapters;

import commands.runnables.moderationcategory.WarnRemoveCommand;
import commands.slashadapters.Slash;
import commands.slashadapters.SlashAdapter;
import commands.slashadapters.SlashMeta;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.CommandData;

@Slash(command = WarnRemoveCommand.class)
public class WarnRemoveAdapter extends SlashAdapter {

    public CommandData addOptions(CommandData commandData) {
        return commandData
                .addOption(OptionType.USER, "member", "The user to lose a specific number of warns", false)
                .addOption(OptionType.USER, "member2", "The user to lose a specific number of warns", false)
                .addOption(OptionType.USER, "member3", "The user to lose a specific number of warns", false)
                .addOption(OptionType.USER, "member4", "The user to lose a specific number of warns", false)
                .addOption(OptionType.USER, "member5", "The user to lose a specific number of warns", false)
                .addOption(OptionType.STRING, "member_id", "The user to lose a specific number of warns", false)
                .addOption(OptionType.STRING, "reason", "The reason of this mod action", false)
                .addOption(OptionType.NUMBER, "amount", "How many warns shall be removed?", false)
                .addOption(OptionType.BOOLEAN, "all", "Remove all warns", false);
    }

    @Override
    public SlashMeta process(SlashCommandEvent event) {
        String args;
        if (event.getOption("all") != null && event.getOption("all").getAsBoolean()) {
            args = collectArgs(event, "amount");
        } else {
            args = collectArgs(event);
        }

        return new SlashMeta(WarnRemoveCommand.class, args);
    }

}

File: src/main/java/commands/slashadapters/adapters/WarnRemoveAdapter.kt
package commands.slashadapters.adapters

import commands.runnables.moderationcategory.WarnRemoveCommand
import commands.slashadapters.Slash
import commands.slashadapters.SlashAdapter
import commands.slashadapters.SlashMeta
import mysql.hibernate.entity.guild.GuildEntity
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData

@Slash(command = WarnRemoveCommand::class)
class WarnRemoveAdapter : SlashAdapter() {

    public override fun addOptions(commandData: SlashCommandData): SlashCommandData {
        return commandData.addOptions(
            generateOptionData(OptionType.STRING, "members", "warnremove_member", true),
            generateOptionData(OptionType.STRING, "reason", "moderation_reason", false),
            generateOptionData(OptionType.INTEGER, "amount", "warnremove_amount", false),
            generateOptionData(OptionType.BOOLEAN, "all", "warnremove_removeall", false)
        )
    }

    override fun process(event: SlashCommandInteractionEvent, guildEntity: GuildEntity): SlashMeta {
        val args: String
        args = if (event.getOption("all")?.asBoolean ?: false) {
            collectArgs(event, "amount")
        } else {
            collectArgs(event)
        }
        return SlashMeta(WarnRemoveCommand::class.java, args)
    }
}